### PR TITLE
Toolbar icons alignment and implement toolbar with different intents

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes https://github.com/zakodium-oss/react-science/issues/590
closes https://github.com/zakodium-oss/react-science/issues/591

### CODE REVIEW

File: src/components/toolbar/Toolbar.tsx [hunk 1]
```diff
@@ -55,45 +44,15 @@ export function Toolbar(props: ToolbarProps) {
...
-      }
-    }
-
-    const element = ref.current;
-    if (element) {
-      const observer = new ResizeObserver(update);
-      observer.observe(element);
-      return () => observer.unobserve(element);
-    }
-  }, [vertical]);
```
Comment (stropitek) [inline src/components/toolbar/Toolbar.tsx:89]: I would be surprised if this workaround is no longer needed.

Please check the vertical layout with wrapping in the stories with Chrome, Firefox and (if possible) Safari.

Comment (wadjih-bencheikh18) [reply src/components/toolbar/Toolbar.tsx:89]: Okay

Comment (wadjih-bencheikh18) [reply src/components/toolbar/Toolbar.tsx:89]: It doesn't work everytime so better keep it

File: src/components/toolbar/Toolbar.tsx [hunk 2]
```diff
@@ -107,30 +105,30 @@ Toolbar.Item = function ToolbarItem(props: ToolbarItemProps) {
...
     <Button
+      {...other}
       minimal
       disabled={disabled}
       css={css`
         .${Classes.ICON} {
           color: ${Colors.DARK_GRAY3};
         }
+        position: relative;
+        font-size: 1.25em;
```
Comment (stropitek) [inline src/components/toolbar/Toolbar.tsx:116]: What are those lines fixing?

File: stories/components/toolbar.stories.tsx [hunk 3]
```diff
@@ -183,6 +184,7 @@ export function HorizontalToolbar() {
             active={state.id === item.id}
             onClick={handleChange}
             icon={item.icon}
+            intent={item.id === 'test5' ? 'danger' : undefined}
```
Comment (stropitek) [inline stories/components/toolbar.stories.tsx:187]: This isn't working. It is still using the intent as passed to the `Toolbar`

File: src/components/toolbar/Toolbar.tsx (hunk 2 above)
Comment (wadjih-bencheikh18) [reply src/components/toolbar/Toolbar.tsx:116]: No I just removed this
```
style={{ position: 'relative', fontSize: '1.25em' }}
```

File: src/components/toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -100,24 +100,28 @@ export function Toolbar(props: ToolbarProps) {
...
 
 Toolbar.Item = function ToolbarItem(props: ToolbarItemProps) {
-  const { active = false, icon, onClick, title, id, ...other } = props;
+  const {
+    active = false,
+    icon,
+    onClick,
+    title,
+    id,
+    intent: intentProps,
```
Comment (stropitek) [inline src/components/toolbar/Toolbar.tsx:109]: Rename to `itemIntent` for clarity

File: src/components/toolbar/Toolbar.tsx [hunk 5]
```diff
@@ -100,24 +100,28 @@ export function Toolbar(props: ToolbarProps) {
...
         .${Classes.ICON} {
           color: ${Colors.DARK_GRAY3};
         }
-        position: relative;
-        font-size: 1.25em;
-        width: 'auto';
       `}
-      intent={intent}
-      // style={{ position: 'relative', fontSize: '1.25em', width: 'auto' }}
+      intent={intentProps || intent}
```
Comment (stropitek) [inline src/components/toolbar/Toolbar.tsx:123]: You can put `intentProps || intent` in another variable since it is used twice.

**Review by stropitek — APPROVED** [review]

**Pull request merged**